codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessage.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+/**
+ * RedisMessage is base interface for codec-redis.
+ */
+public interface RedisMessage {
+
+    /**
+     * Returns the {@link RedisMessageType}.
+     *
+     * @return the message type.
+     */
+    RedisMessageType type();
```

Comment:
At first commit of this PR, I don't want to complex type hierachy for Redis messages. So, I did it with only one `RedisMessage` concrete class with `type()` method. But now, it's true that type hierachy and `type()` are redundant. And, It's not necessary if we seperate `StringRedisMessage` to 2 types(SIMPLE_STRING and ERROR). @mp911de and I tried to merge type hierachy, but It didn't merge well. Anyway, I'm not sure seperating is good or not. WDYT?
(If we'll seperate `StringRedisMessage` to 2 types, I want to remove factory too.)